Release 2.3.0 (#6995)
antrea-io/antrea: CHANGELOG/CHANGELOG-2.3.md

# Changelog 2.3

## 2.3.0 - 2025-02-19

### Added

- Add `Proxy` mode for Flow Aggregator to send flows directly without buffering or aggregation. ([#6920](https://github.com/antrea-io/antrea/pull/6920) [#6961](https://github.com/antrea-io/antrea/pull/6961), [@antoninbas])

> - Support version skew between Antrea Agent and Flow Aggregator to improve upgrade robustness. ([#6912](https://github.com/antrea-io/antrea/pull/6912), [@antoninbas])

Reviewer: ```suggestion
- Support version skew between antrea-agent and `FlowAggregator` to improve upgrade robustness. ([#6912](https://github.com/antrea-io/antrea/pull/6912), [@antoninbas])
```

Reviewer: So do the following ones.

Author: I would prefer to `Antrea Agent` and `Flow Aggregator` in the release log. I feel antrea-agent is used when it's specific to the process.

> - Add `clusterId` to aggregated records for Flow Aggregator. ([#6769](https://github.com/antrea-io/antrea/pull/6769), [@antoninbas])

Reviewer: ```suggestion
- Add `clusterId` to aggregate records for Flow Aggregator. ([#6769](https://github.com/antrea-io/antrea/pull/6769), [@antoninbas])
```

Author: “aggregated” is an objective here.

# Changelog 2.3

## 2.3.0 - 2025-02-21

### Added

- Add `Proxy` mode for Flow Aggregator to send flows directly without buffering or aggregation. ([#6920](https://github.com/antrea-io/antrea/pull/6920) [#6961](https://github.com/antrea-io/antrea/pull/6961), [@antoninbas])
- Support version skew between Antrea Agent and Flow Aggregator to improve upgrade robustness. ([#6912](https://github.com/antrea-io/antrea/pull/6912), [@antoninbas])
- Add `clusterId` to aggregated records for Flow Aggregator. ([#6769](https://github.com/antrea-io/antrea/pull/6769), [@antoninbas])
- Add `checksum/config` annotation to the Deployment of Flow Aggregator. ([#6967](https://github.com/antrea-io/antrea/pull/6967), [@antoninbas])

> - Add more fields in `BGPPolicy` API to support BGP confederation. ([#6905](https://github.com/antrea-io/antrea/pull/6905), [@hongliangl])

Reviewer 2: I wonder if we should omit this changelog entry given that we have postponed implementation to v2.4
It may create some confusion

Author: Removed it, I will include it in 2.4 release log.

- Support SecondaryNetwork of `SR-IOV` type for VM Nodes. ([#6881](https://github.com/antrea-io/antrea/pull/6881), [@tnqn])
- Add more printer columns for `PacketCapture` CRD. ([#6977](https://github.com/antrea-io/antrea/pull/6977), [@antoninbas])

> - Add logs collection for failed agents and controller for Support Bundle. ([#3659](https://github.com/antrea-io/antrea/pull/3659), [@hangyan])

Reviewer 2: Add fallback log collection to the `antctl supportbundle` command for Antrea components for which regular Support Bundle collection has failed.

- Add `antreaProxy.disableServiceHealthCheckServer` config to disable the health check server run by Antrea Proxy to avoid `kube-proxy` error logs. ([#6939](https://github.com/antrea-io/antrea/pull/6939), [@antoninbas])

> - Add route info in the output of `antctl get bgproutes`. ([#6803](https://github.com/antrea-io/antrea/pull/6803) [#6835](https://github.com/antrea-io/antrea/pull/6835), [@Atish-iaf])

Reviewer 2: to the output

### Changed

- Promote feature `EgressSeparateSubnet` from Alpha to Beta. ([#6982](https://github.com/antrea-io/antrea/pull/6982), [@luolanzone])
- Promote feature `ServiceExternalIP` from Alpha to Beta. ([#6903](https://github.com/antrea-io/antrea/pull/6903), [@xliuxu])

> - Accept configurations with no collector / sink to avoid crash issue for Flow Aggregator. ([#7006](https://github.com/antrea-io/antrea/pull/7006), [@antoninbas])

Reviewer 3: how about "Allow running Flow Aggregator with no collector / sink."

> - More efficient IP check to query `NodeRouteController` for storing all the PodCIDRs to determine whether the source / destination IPs are Pod IPs (NodeIPAM only) for `FlowExporter`. ([#6960](https://github.com/antrea-io/antrea/pull/6960), [@antoninbas])

Reviewer 2: I think this is too confusing for users and `NodeRouteController` is an implementation "detail" here.
Maybe:

> More efficient IP checks in the Flow Exporter when determining the type of flow being exported.

- Remove stale OVS interfaces in the CNIServer reconciler if the original Pod interface is disconnected. ([#6919](https://github.com/antrea-io/antrea/pull/6919), [@wenyingd])
- Remove local ASN range limitation in `BGPPolicy` API. ([#6914](https://github.com/antrea-io/antrea/pull/6914), [@hongliangl])
- Support providing a fixed public host key for SFTP uploads with a new field `hostPublicKey` to `PacketCapture` and `SupportBundleCollection` CRDs. ([#6848](https://github.com/antrea-io/antrea/pull/6848), [@antoninbas])

> - Skip processing Service without IPPool annotation `service.antrea.io/external-ip-pool` for `ServiceExternalIP` feature. ([#6950](https://github.com/antrea-io/antrea/pull/6950), [@xliuxu])

Reviewer 3: This doesn't have real impact to users, maybe no need to mention.

Author: Removed

> - Improve the document of the configuration option `egress.exceptCIDRs`. ([#6828](https://github.com/antrea-io/antrea/pull/6828), [@tnqn])

Reviewer 3: this is a minor doc change, no need to mention

- Upgrade CNI plugins from v1.5.1 to v1.6.2. ([#6796](https://github.com/antrea-io/antrea/pull/6796), [@luolanzone])
- Push Antrea Ubuntu-based images to `ghcr.io`. ([#6834](https://github.com/antrea-io/antrea/pull/6834), [@antoninbas])

> - Upgrade go-ipfix to 0.13.0. ([#6998](https://github.com/antrea-io/antrea/pull/6998), [@antoninbas])

Reviewer 2: Upgrade go-ipfix to 0.13.0, which includes performance improvements and supports sending multiple data records in the same IPFIX message.

### Fixed

> - Add `-ComputerName localhost` explicitly for VMSwitch commands to avoid potential validation issue on Windows with Active Directory. ([#6985](https://github.com/antrea-io/antrea/pull/6985), [@XinShuYang])

Reviewer 2: "a validation issue" or "validation issues"

- Fix that Antrea L7NetworkPolicies do not handle Service traffic correctly. ([#6941](https://github.com/antrea-io/antrea/pull/6941), [@hongliangl])
- Disable TX checksum offload for Antrea host gateway interface when `disableTXChecksumOffload` is set to `true`. ([#6843](https://github.com/antrea-io/antrea/pull/6843), [@hongliangl])

> - Add `fqdnCacheMinTTL` configuration for Antrea ClusterNetworkPolicy to ensure that resolved IPs are included in data path rules as long as the application caches them. ([#6808](https://github.com/antrea-io/antrea/pull/6808), [@hkiiita])

Reviewer 3: This is not specific to ClusterNetworkPolicy, s/Antrea ClusterNetworkPolicy/Antrea-native policies/

> - Add `fqdnCacheMinTTL` configuration for Antrea ClusterNetworkPolicy to ensure that resolved IPs are included in data path rules as long as the application caches them. ([#6808](https://github.com/antrea-io/antrea/pull/6808), [@hkiiita])

Reviewer 2: which will ensure that resolved IPs are included in data path rules for at least the configured amount of time, in case some applications are caching the results of DNS queries

> - Support OF port state "down" when installing forwarding rules for Pods on Windows. ([#6889](https://github.com/antrea-io/antrea/pull/6889), [@wenyingd])

Reviewer 3: hard to get the user impact from the description, maybe describe it as:
Ensure that OpenFlow rules for a Windows Pod are installed as long as the OpenFlow port is allocated, even if its state is incorrectly reported as "LINK_DOWN".

> - Fix audit logging for default deny all K8s NetworkPolicy rules. ([#6855](https://github.com/antrea-io/antrea/pull/6855), [@qiyueyao])

Reviewer 3: ```suggestion
- Fix audit logging for default deny-all K8s NetworkPolicy rules. ([#6855](https://github.com/antrea-io/antrea/pull/6855), [@qiyueyao])
```

- Ensure that `promote_secondaries` is set on `IPAssigner` interfaces to avoid the automatic removal of all other IP addresses in the same subnet when the primary IP address is deleted. ([#6898](https://github.com/antrea-io/antrea/pull/6898) [#6900](https://github.com/antrea-io/antrea/pull/6900), [@antoninbas])

> - Set the maximum packet size explicitly to fix an issue with reading `PacketCapture` pcapng files in `tcpdump` on macOS. ([#6804](https://github.com/antrea-io/antrea/pull/6804), [@hangyan])

Reviewer 2: with `tcpdump`

- Reconcile Pods with `hostNetwork` after Antrea Agent is restarted on Windows. ([#6944](https://github.com/antrea-io/antrea/pull/6944), [@wenyingd])
- Create a new kubeconfig for `SupportBundleClient` to fix `antctl supportbundle` failures on Windows. ([#6840](https://github.com/antrea-io/antrea/pull/6840), [@XinShuYang])

> - Fix race condition when getting BGP routes in BGPController. ([#6823](https://github.com/antrea-io/antrea/pull/6823), [@Atish-iaf])

Reviewer 2: maybe we could omit this since the functionality is new in this release

alternatively, you can add this PR number to the "Add route info in the output of `antctl get bgproutes`" entry

Author: Move it with the "Add route info in the output of antctl get bgproutes" entry.
